### ISSUE TITLE
`misc-const-correctness.WarnPointersAsValues` not reported when accessed through `->`

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#56777]
```cpp
class A
{
public:
    void f();
};

extern A* cb1();
extern void cb2(A*);

void f()
{
    A* a1 = cb1(); // no warning
    a1->f();

    A* a2 = cb1(); // warning
    cb2(a2);
}
```
https://godbolt.org/z/qde3M79fY

**Comment by EugeneZelenko** [comment]
@JonasToth
